url708d477f92_zSakare_SENG3011_tgz-pJ8-test_runner_TestRunnerJ8/src/main/implementations/order/Order.java

```diff
@@ -0,0 +1,204 @@
+	 */
+	public String getInstrument() {
+		return instrument;
+	}
+	public void setInstrument(String instrument) {
+		this.instrument = instrument;
+	}
+	public Date getDateTime() {
+		return dateTime;
+	}
+	public void setDateTime(Date dateTime) {
+		this.dateTime = dateTime;
+	}
+	public String getRecordType() {
+		return recordType;
+	}
+	public void setRecordType(String recordType) {
+		this.recordType = recordType;
+	}
+	public double getPrice() {
+		return price;
+	}
+	public void setPrice(double price) {
+		this.price = price;
+	}
+	public int getVolume() {
+		return volume;
+	}
+	public void setVolume(int volume) {
+		this.volume = volume;
+	}
+	public int getUndisclosedVolume() {
+		return undisclosedVolume;
+	}
+	public void setUndisclosedVolume(int undisclosedVolume) {
+		this.undisclosedVolume = undisclosedVolume;
+	}
+	public double getValue() {
+		return value;
+	}
+	public void setValue(double value) {
+		this.value = value;
+	}
+	public String getQualifiers() {
+		return qualifiers;
+	}
+	public void setQualifiers(String qualifiers) {
+		this.qualifiers = qualifiers;
+	}
+	public String getTransactionId() {
+		return transactionId;
+	}
+	public void setTransactionId(String transactionId) {
+		this.transactionId = transactionId;
+	}
+	public String getBidId() {
+		return bidId;
+	}
+	public void setBidId(String bidId) {
+		this.bidId = bidId;
+	}
+	public String getAskId() {
+		return askId;
+	}
+	public void setAskId(String askId) {
+		this.askId = askId;
+	}
+	public boolean isBid() {
+		return isBid;
+	}
+	public void setBid(boolean isBid) {
+		this.isBid = isBid;
+	}
+	public Date getEntryTime() {
+		return entryTime;
+	}
+	public void setEntryTime(Date entryTime) {
+		this.entryTime = entryTime;
+	}
+	public double getOldPrice() {
+		return oldPrice;
+	}
+	public void setOldPrice(double oldPrice) {
+		this.oldPrice = oldPrice;
+	}
+	public int getOldVolume() {
+		return oldVolume;
+	}
+	public void setOldVolume(int oldVolume) {
+		this.oldVolume = oldVolume;
+	}
+	public String getBuyerBrokerId() {
+		return buyerBrokerId;
+	}
+	public void setBuyerBrokerId(String buyerBrokerId) {
+		this.buyerBrokerId = buyerBrokerId;
+	}
+	public String getSellerBrokerId() {
+		return sellerBrokerId;
+	}
+	public void setSellerBrokerId(String sellerBrokerId) {
+		this.sellerBrokerId = sellerBrokerId;
+	}
+	
+	public boolean equals(Object o) {
+		boolean equals = false;
+		
+		if (o != null) {
+			Order order = (Order) o;
+			if (this.isBid == order.isBid
+					&& this.askId == order.askId
+					&& this.bidId == order.bidId
+					&& this.buyerBrokerId == order.buyerBrokerId
+					&& this.dateTime == order.dateTime
+					&& this.entryTime == order.entryTime
+					&& this.instrument == order.instrument
+					&& this.oldPrice == order.oldPrice
+					&& this.oldVolume == order.oldVolume
+					&& this.price == order.price
+					&& this.qualifiers == order.qualifiers
+					&& this.recordType == order.recordType
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.